pandas/core/apply.py

```diff
@@ -1427,14 +1427,17 @@ def _managle_lambda_list(aggfuncs: Sequence[Any]) -> Sequence[Any]:
     if len(aggfuncs) <= 1:
         # don't mangle for .agg([lambda x: .])
         return aggfuncs
-    i = 0
+
     mangled_aggfuncs = []
-    for aggfunc in aggfuncs:
-        if com.get_callable_name(aggfunc) == "<lambda>":
-            aggfunc = partial(aggfunc)
-            aggfunc.__name__ = f"<lambda_{i}>"
-            i += 1
-        mangled_aggfuncs.append(aggfunc)
+    for idx, aggfunc in enumerate(aggfuncs):
+        if not com.get_callable_name(aggfunc) == "<lambda>":
+            mangled_aggfuncs.append(aggfunc)
+            continue
+
+        partial_aggfunc = aggfunc
```

Comment:
@MarcoGorelli by adding this line the error should be solved. At least locally the mypy error is gone. Not sure why but by simply creating `partial_aggfunc` from the call to `partial(aggfunc)` in the next line it removes the `__name__` attribute. But if I firstly assign it as equal to `aggfunc` as I am doing here it works fine. Do you know any similar behavior?

Reply:
Oh thank you a lot for the suggestion! I will implement it like this. Sorry but I am still learning how to properly handle those issues.